packages/data-stores/src/woocommerce-eligibility/test/reducer.ts

```diff
@@ -0,0 +1,32 @@
+import { transferEligibility } from '../reducer';
+import { TransferEligibility } from '../types';
+
+describe( 'reducer', () => {
+	it( 'returns the correct default state', () => {
+		const state = transferEligibility( undefined, { type: 'TEST_ACTION' } );
+		expect( state ).toEqual( undefined );
+	} );
+
+	it( 'returns the site transfer elibibility', () => {
+		const siteId = 1234;
+		const expectedTransferEligibility: TransferEligibility = {
+			errors: [],
+			warnings: {
+				plugins: [],
+				widgets: [],
+				subdomain: [],
+			},
+			is_eligible: false,
+		};
+
+		const state = transferEligibility( undefined, {
+			type: 'TRANSFER_ELIGIBILITY_RECEIVE',
+			siteId,
+			transferEligibility: expectedTransferEligibility,
+		} );
+
+		expect( state ).toEqual( {
+			[ siteId ]: expectedTransferEligibility,
```

Comment:
When testing such reducers, in addition to a test that ensures that keying by site ID works, we could also benefit from testing those scenarios:

* Data is updated properly when passed for a site ID that is already in the state.
* Data is added properly for a new site, while data for existing site IDs is kept intact.